### PR TITLE
Add support for P4estMesh

### PR DESCRIPTION
**Pull request by efaulhaber**
grafik
grafik
grafik
grafik

### CODE REVIEW

File: src/interpolate.jl [hunk 1]
```diff
@@ -11,7 +11,7 @@ end
 
 # Interpolate data from input format to desired output format (CurvedMesh or UnstructuredQuadMesh version)
 function interpolate_data(::Val{:vtu}, input_data,
-                          mesh::Union{Trixi.CurvedMesh, Trixi.UnstructuredQuadMesh},
+                          mesh::Union{Trixi.CurvedMesh, Trixi.UnstructuredQuadMesh, Trixi.P4estMesh},
```
Comment (sloede) [inline src/interpolate.jl:14]: If you need these mesh types in multiple places in Trixi2Vtk, you could change
```julia
import Trixi
```
to
```julia
using Trixi: CurvedMesh, UnstructuredQuadMesh, P4estMesh, TreeMesh
```
in `Project.toml`.

Comment (efaulhaber) [reply src/interpolate.jl:14]: Good point!

**Comment by efaulhaber** [comment]
This is basically done, but the `P4estMesh` needs to be in a Trixi release first, so that we can update the required version here.

File: src/vtktools.jl [hunk 2]
```diff
@@ -164,6 +166,20 @@ function calc_node_coordinates(mesh::Trixi.UnstructuredQuadMesh, n_visnodes)
...
+  # Extract number of spatial dimensions
+  ndims_ = ndims(mesh)
+
+  nodes = range(-1, 1, length=n_visnodes)
+
+  node_coordinates = Array{Float64, ndims_+2}(undef, ndims_,
+                                              ntuple(_ -> n_visnodes, ndims_)...,
+                                              Trixi.ncells(mesh))
+
+  return Trixi.calc_node_coordinates!(node_coordinates, mesh, nodes)
```
Comment (sloede) [inline src/vtktools.jl:179]: Is `calc_node_coordinates!` correct here or should it be
```suggestion
  return Trixi.calc_tree_node_coordinates!(node_coordinates, mesh, nodes)
```
?

File: test/test_2d.jl [hunk 3]
```diff
@@ -127,6 +127,26 @@ end
       end
     end
   end
+
+  @testset "P4estMesh" begin
+    isdir(outdir) && rm(outdir, recursive=true)
+    run_trixi(joinpath("2d", "elixir_euler_nonperiodic_p4est.jl"), initial_refinement_level=1, maxiters=1)
```
Comment (sloede) [inline test/test_2d.jl:133]: Have you tested this for higher refinement levels as well?

Comment (sloede) [reply test/test_2d.jl:133]: Or does it make sense to include a specific test for it (or just make it the default here, e.g., `2`)?

**Review by sloede — CHANGES_REQUESTED** [review]
LGTM! Just a 2-3 questions

File: test/test_2d.jl (hunk 3 above)
Comment (efaulhaber) [reply test/test_2d.jl:133]: I tested it for levels 0, 1, and 2. But Trixi2Vtk doesn't use the level or the trees at all. The `node_coordinates` of each element are saved in the element container the same way as with `CurvedMesh`.

Comment (sloede) [reply test/test_2d.jl:133]: Ah, right: level 1 is already one higher then the base level. In this case we can just leave it as it is, I think

File: src/vtktools.jl (hunk 2 above)
Comment (efaulhaber) [reply src/vtktools.jl:179]: No. These are element node coordinates, not tree node coordinates. That's why I renamed all tree related stuff, that became a bit confusing.

Comment (sloede) [reply src/vtktools.jl:179]: Got it, thanks

**Review by sloede — APPROVED** [review]

**Pull request merged**